src/python/pants/binaries/binary_tool.py

```diff
@@ -70,10 +84,35 @@ def select(self, context=None):
         version = old_opts.get(self.replaces_name)
     return self._select_for_version(version)
 
+  @memoized_property
+  def _binary_util(self):
+    return BinaryUtil.Factory.create()
+
+  # can override this and call super() to compose
```

Comment:
Nit: Complete sentences imply complete thoughts. So begin a sentence with a capital letter, end it with a period.

Reply:
This was also removed in 27d0c37, but I will also keep this in mind.